### PR TITLE
[SYCL][joint matrix spec] clarify coodinates of packed matrix and fix joint_matrix_copy requirements

### PR DESCRIPTION
**Pull request by dkhaldi**

### CODE REVIEW

**Comment by dkhaldi** [comment]
@gmlueck, gentle ping to review/approve.

File: sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_oneapi_matrix.asciidoc [hunk 1]
```diff
@@ -317,18 +317,18 @@ void joint_matrix_fill(Group g, joint_matrix<Group, T, Use, Rows,
 namespace sycl::ext::oneapi::experimental::matrix {
 
 template <typename Group, typename T1, typename T2, size_t Rows,
-          size_t Cols, use Use1, use Use2, layout Layout> 
+          size_t Cols, use Use1, use Use2, layout Layout1, layout Layout2>
 void joint_matrix_copy(Group g,
-                       joint_matrix<Group, T1, Use1, Rows, Cols, Layout> &src,
-                       joint_matrix<Group, T2, Use2, Rows, Cols, Layout> &dest);
+                      joint_matrix<Group, T1, Use1, Rows, Cols, Layout1> &dest,
+                      joint_matrix<Group, T2, Use2, Rows, Cols, Layout2> &src);
```
[the remark this answers is not in the document]
Comment (dkhaldi) [reply sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_oneapi_matrix.asciidoc:323]: You are right. The pdf version I use (SYCL 2020 rev 2)  used the opposite. But I think it is an old version (see below). I will switch things back. 

/* -- USM functions -- */
116
117 event memcpy(void* dest, const void* src, size_t numBytes);
118 event memcpy(void* dest, const void* src, size_t numBytes,
119 event depEvent);
120 event memcpy(void* dest, const void* src, size_t numBytes,
121 const std::vector<event> &depEvents);
4.6.5.1. Queue interface SYCL 2020 rev 2
88 | Chapter 4. SYCL programming interface122
123 template <typename T>
124 event copy(T* dest, const T *src, size_t count);
125 template <typename T>
126 event copy(T* dest, const T *src, size_t count,
127 event depEvent);
128 template <typename T>
129 event copy(T* dest, const T *src, size_t count,
130 const std::vector<event> &depEvents);

**Review by gmlueck — APPROVED** [review]

**Comment by dkhaldi** [comment]
@intel/llvm-gatekeepers, please help merge

File: sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_intel_matrix.asciidoc [hunk 2]
```diff
@@ -198,6 +198,11 @@ joint_matrix_apply(sg, A, [=](T &val, size_t row, size_t  col) {
    sum_local_rows[row] += val;
 });
 ```
+
+In the case of `ext_intel_packed` matrix memory layout, `row` and
```
Comment (MrSidims) [inline sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_intel_matrix.asciidoc:202]: I believe this should be also reflected in SPIR-V specification for coordinate instruction. @dkhaldi @Nuullll

Comment (dkhaldi) [reply sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_intel_matrix.asciidoc:202]: I agree

**Pull request merged**